extra/entrypoint.sh

```diff
@@ -2,20 +2,54 @@
 
 # set -e Exit the script if an error happens
 set -e
-PUID=${PUID=0}
-PGID=${PGID=0}
+
 
 files_ownership () {
-    # -h Changes the ownership of an encountered symbolic link and not that of the file or directory pointed to by the symbolic link.
-    # -R Recursively descends the specified directories
-    # -c Like verbose but report only when a change is made
-    chown -hRc "$PUID":"$PGID" /app/data
+    # Check if the /app/data folder is owned by the user invoking the container
+    if [ $(stat -c%u /app/data) !=  $(id -u) ]; then
+        echo "File ownership incorrect, attempting to fix."
+        chown -hRc "$(id -u)":"$(id -g)" /app/data || echo "ERROR: Failed to set file ownership. Please run 'sudo chown -R $(id -u):$(id -g) /path/to/container/volume' to resolve."; exit 1
+    fi
+
+   # Checks for R/W permissions
+   if [ $(stat -c%a /app/data) -ne 770 ]; then
+        echo "Directory permissions incorrect, attempting to fix."
+        find /app/data -type d -exec chmod 770 {} \;
+
+        #Re-run the check
+        if [ $(stat -c%a /app/data) -ne 770 ]; then
+                echo "ERROR: Failed to set file permissions. Please run 'sudo find /path/to/container/volume -type d chmod 770 {} \;' to resolve."
+                exit 1
+        fi
+        echo "Directory permission fix succesful! Continuing."
+   fi
+
+   #Check the R/W permissions on the files
+   if [ $(stat -c%a /app/data/* | head -n 1) != 640 ]; then
+        echo "File permissions incorrect. Attempting to fix."
+        find /app/data -type f -exec chmod 640 {} \;
+
+        #Re-run the check
```

Comment:
Be consistent in comments
```suggestion
        # Re-run the check
```